src/sonic-config-engine/tests/test_multinpu_cfggen.py

```diff
@@ -0,0 +1,229 @@
+import unittest
+from unittest import TestCase
+import subprocess
+import os
+import json
+import yaml
+
+SKU = 'multi-npu-01'
+ASIC_SKU = 'multi-npu-asic'
+NUM_ASIC = 4
+HOSTNAME = 'multi_npu_platform_01'
+
+
+class TestMultiNpuCfgGen(TestCase):
+
+    def setUp(self):
+        self.test_dir = os.path.dirname(os.path.realpath(__file__))
+        self.test_data_dir = os.path.join(self.test_dir,  'multi_npu_data')
+        self.script_file = os.path.join(self.test_dir, '..', 'sonic-cfggen')
+        self.sample_graph = os.path.join(self.test_data_dir, 'sample-minigraph.xml')
+        self.port_config = []
+        for asic in range(NUM_ASIC):
+            self.port_config.append(os.path.join(self.test_data_dir, "sample_port_config-{}.ini".format(asic)))
+
+
+    def run_script(self, argument, check_stderr=False):
+        print '\n    Running sonic-cfggen ' + argument
+        if check_stderr:
+            output = subprocess.check_output(self.script_file + ' ' + argument, stderr=subprocess.STDOUT, shell=True)
+        else:
+            output = subprocess.check_output(self.script_file + ' ' + argument, shell=True)
+
+        linecount = output.strip().count('\n')
+        if linecount <= 0:
+            print '    Output: ' + output.strip()
+        else:
+            print '    Output: ({0} lines, {1} bytes)'.format(linecount + 1, len(output))
+        return output
+
+    def run_diff(self, file1, file2):
+        return subprocess.check_output('diff -u {} {} || true'.format(file1, file2), shell=True)
+
+
+    def run_script_for_asic(self,argument,asic, port_config=None):
+        argument = "{} -n asic{} ".format(argument, asic)
+        if port_config:
+            argument += "-p {}".format(port_config)
+        output = self.run_script(argument)
+        return output
+
+    def test_dummy_run(self):
+        argument = ''
+        output = self.run_script(argument)
+        self.assertEqual(output, '')
+    
+    def test_hwsku(self):
+        argument = "-v \"DEVICE_METADATA[\'localhost\'][\'hwsku\']\" -m \"{}\"".format(self.sample_graph)
+        output = self.run_script(argument)
+        self.assertEqual(output.strip(), SKU)
+        
+        for asic in range(NUM_ASIC):
+            output = self.run_script_for_asic(argument, asic)
+            self.assertEqual(output.strip(), SKU)
+        
+
```

Comment:
inconsistent while lines between functions, sometime two lines, sometimes one line.

Reply:
Fixed in the latest commit